### PR TITLE
1021 Refactor Pet instance method to use the Match model instead of the AdopterApplication model

### PR DESCRIPTION
**Pull request by coalest**
# 🔗 Issue
#1021 

# ✍️ Description
- Refactored `Pet#is_adopted?` method to reference `Match` association rather than `AdopterApplication`
- Fixed related tests

### CODE REVIEW

**Comment by coalest** [comment]
I'm sure that performance isn't a priority at the moment, but I noticed while refactoring the `is_adopted?` method, that there is an N+1 on the `/staff/pets` page because of this method and the `in_foster?`/`open?` methods are called in order to fetch the status of every pet.

Before changes:
Screenshot from 2024-10-10 16-29-50

After changes:
Screenshot from 2024-10-10 16-30-58

You can see the number of Active Record queries goes down significantly. I can put this into a separate PR, or just stash these changes for now, if we don't want to look at performance yet.

File: app/controllers/organizations/adoptable_pets_controller.rb [hunk 1]
```diff
@@ -8,10 +8,9 @@ class Organizations::AdoptablePetsController < Organizations::BaseController
   helper_method :get_animals
 
   def index
-    @q = authorized_scope(Pet.includes(:adopter_applications, images_attachments: :blob),
-      with: Organizations::AdoptablePetPolicy).ransack(params[:q])
+    @q = authorized_scope(Pet.all, with: Organizations::AdoptablePetPolicy).ransack(params[:q])
```
Comment (kasugaijin) [inline app/controllers/organizations/adoptable_pets_controller.rb:11]: Do you think we should only return pets that are unadopted for this collection?

**Review by kasugaijin — APPROVED** [review]
Nice one! Looks like you were N+1 hunting

File: app/controllers/organizations/adoptable_pets_controller.rb (hunk 1 above)
Comment (coalest) [reply app/controllers/organizations/adoptable_pets_controller.rb:11]: Sure, I can do that. Okay if I add that to this PR?

Comment (coalest) [reply app/controllers/organizations/adoptable_pets_controller.rb]: Added to this PR for now (see last commit).

Comment (kasugaijin) [reply app/controllers/organizations/adoptable_pets_controller.rb]: Yes in this PR is fine. Thanks.

**Review by kasugaijin — APPROVED** [review]
LGTM!

**Pull request merged**